src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -824,10 +831,13 @@ public int ExecuteOnThread(IDebugThread2 pThread)
         // that is, not all threads should be required to be stopped before this method returns. The implementation of this method may be 
         // as simple as calling the IDebugProgram2::CauseBreak method on this program.
         //
-        // The sample engine only supports debugging native applications and therefore only has one program per-process
         public int Stop()
         {
-            throw new NotImplementedException();
+            DebuggedProcess.WorkerThread.RunOperation(async () =>
+            {
+                await _debuggedProcess.CmdBreak();
```

Comment:
The problem is that for the attach case, we send the debuggee a signal and the api to send the signal (kill) returns before the debuggee actually processes the signal. There is a known bug because of that since the await will return early. There are no scenarios were really depending on the timing, but this one might actually need to be sure the debuggee is stopped